codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientClass.java

```diff
@@ -164,16 +165,45 @@ private MethodSpec constructorWithAdvancedConfiguration() {
     }
 
     private List<MethodSpec> operations() {
-        return model.getOperations().values().stream().map(this::operationMethodSpec).collect(Collectors.toList());
+        return model.getOperations().values().stream()
+                    .map(this::operationMethodSpec)
+                    .flatMap(List::stream)
+                    .collect(Collectors.toList());
     }
 
-    private MethodSpec operationMethodSpec(OperationModel opModel) {
-        return SyncClientInterface.operationMethodSignature(model, opModel)
+    private List<MethodSpec> operationMethodSpec(OperationModel opModel) {
+        List<MethodSpec> methods = new ArrayList<>();
+
+        if (opModel.getInputShape().isSimpleMethod()) {
+            methods.add(simpleMethod(opModel));
+        }
+
+        methods.add(SyncClientInterface.operationMethodSignature(model, opModel)
                                   .addAnnotation(Override.class)
                                   .addCode(protocolSpec.responseHandler(opModel))
                                   .addCode(protocolSpec.errorResponseHandler(opModel))
                                   .addCode(protocolSpec.executionHandler(opModel))
-                                  .build();
+                                  .build());
+
+        return methods;
+    }
+
+    private MethodSpec simpleMethod(OperationModel opModel) {
+        MethodSpec.Builder builder = SyncClientInterface.operationSimpleMethodSignature(model, opModel)
+                           .addAnnotation(Override.class)
+                           .addCode("return $N($N.builder().build()",
+                                         opModel.getMethodName(),
+                                         opModel.getInput().getVariableType());
+
+        if (opModel.hasStreamingInput()) {
```

Comment:
Will update to do so. All should be ignored anyways as it'd be weird to not have a required arg on a streaming op.